src/sdam/topology.ts

```diff
@@ -464,20 +464,14 @@ export class Topology extends TypedEventEmitter<TopologyEvents> {
     };
 
     try {
-      const server = await this.selectServer(
```

Comment:
Why is this change necessary?

Reply:
This introduces latency while the driver attempts to discover the topology for a different read preference than the auto-connecting operation is using. Since this selectServer uses [MongoClient ReadPreference or default primary](https://github.com/mongodb-js/node-mongodb-native/blob/a7feb2c07347176d0e6c0ff2468218700ef6e1dc/src/sdam/topology.ts#L453) and the operation can use [secondary or specific tagSets](https://github.com/mongodb-js/node-mongodb-native/blob/a7feb2c07347176d0e6c0ff2468218700ef6e1dc/src/operations/execute_operation.ts#L90) we block in auto-connect for longer than needed or end up throwing `MongoServerSelectionError` for an operation that may have a selectable server.